src/main/java/com/teragrep/pth_06/planner/BloomFilterTempTable.java

```diff
@@ -0,0 +1,236 @@
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner;
+
+import com.teragrep.blf_01.Token;
+import org.apache.spark.util.sketch.BloomFilter;
+import org.jooq.*;
+import org.jooq.impl.DSL;
+import org.jooq.types.ULong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Set;
+
+import static com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb.BLOOMDB;
+import static org.jooq.impl.SQLDataType.BIGINTUNSIGNED;
+
+/**
+ * Class to represent a SQL Temp. Filled with bloom filters of all the parent tables records filter types. Search term
+ * token set parameter is inserted into each filter. Used to generate bloommatch condition that compares this filter
+ * column against parent table filter column row by row using bloommatch UDF.
+ * <p>
+ * Parent Table Schema:
+ * <ol>
+ * <li>id PK</li>
+ * <li>partition_id FK journaldb.logfile.id</li>
+ * <li>filter_type_id FK bloomdb.filtertype.id</li>
+ * <li>filter - bloomfilter byte array</li>
+ * </ol>
+ * <p>
+ * Temp Table Schema:
+ * <ol>
+ * <li>id PK</li>
+ * <li>termId - count of the current search term</li>
+ * <li>typeId FK bloomdb.filtertype.id</li>
+ * <li>filter - bloomfilter bytes with only the search term token set inserted</li>
+ * </ol>
+ * Parent table create table example:
+ * <p>
+ * <code>
+ *     CREATE TABLE `example` ( `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT, `partition_id` bigint(20) unsigned NOT NULL, `filter_type_id` bigint(20) unsigned NOT NULL, `filter` longblob NOT NULL, PRIMARY KEY (`id`), UNIQUE KEY `partition_id` (`partition_id`), KEY `example_ibfk_1` (`filter_type_id`), CONSTRAINT `example_ibfk_1` FOREIGN KEY (`filter_type_id`) REFERENCES `filtertype` (`id`) ON DELETE CASCADE, CONSTRAINT `example_ibfk_2` FOREIGN KEY (`partition_id`) REFERENCES `journaldb`.`logfile` (`id`) ON DELETE CASCADE ) ENGINE=InnoDB AUTO_INCREMENT=54787 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
+ * </code>
+ *
+ * @see com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb
+ */
+public final class BloomFilterTempTable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BloomFilterTempTable.class);
+
+    private final DSLContext ctx;
+    private final Table<?> parentTable;
+    private final long bloomTermId;
+    private final Set<Token> tokenSet;
+
+    public BloomFilterTempTable(DSLContext ctx, Table<?> parentTable, long bloomTermId, Set<Token> tokenSet) {
+        this.ctx = ctx;
+        this.parentTable = parentTable;
+        this.bloomTermId = bloomTermId;
+        this.tokenSet = tokenSet;
```

Comment:
Perhaps this could be named as searchTokens or similar? to clarify which tokens are used.

Reply:
renamed to searchTermTokenSet